File: src/MultilingualFileProcessingPlatform.Api/Models/Job.cs
namespace MultilingualFileProcessingPlatform.Api.Models;

/// <summary>
/// Represents a file-processing job.
/// </summary>
public class Job
{
    /// <summary>
    /// Unique identifier for the job.
    /// </summary>
    public Guid Id { get; set; }

    /// <summary>
    /// User-friendly name for the job.
    /// </summary>
    public string Name { get; set; } = string.Empty;

    /// <summary>
    /// Current processing status of the job.
    /// </summary>
    public string Status { get; set; } = string.Empty;

    /// <summary>
    /// Date and time when the job was created.
    /// </summary>
    public DateTime CreatedAt { get; set; }
}

File: frontend/src/app/models/job.ts
export interface Job {
  id: string;
  name: string;
  status: string;
  createdAt: string;
}

export interface TranslationValidation {
  isValid: boolean;
  missingSegmentIds: string[];
  duplicateSegmentIds: string[];
  unexpectedSegmentIds: string[];
}